### ISSUE TITLE
progress bar resets partway through import

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri#2206]
This has been known for quite some time, but I don't think we have an issue actually tracking the issue. Please close as a duplicate if that's not the case.

When importing a channel, the progress bar gets to 100% and then resets to 0% partway through the import process. These stages correspond to multiple sub-tasks in Iceqube.

cc @aronasorman

**Comment by jonboiser** [comment]
This does look like a duplicate of #1927.

**Comment by indirectlylit** [comment]
yeah they definitely look related. Wasn't sure if they were the same because #1927 references "multiple channels" whereas this one is just one channel

**Comment by jonboiser** [comment]
@aronasorman Sometimes when importing, there is a Task that follows the `scheduled` and `queued` tasks with both `status = running` and `percentage = 1`. Then right after, the percentage goes back to something < 1. This is what causes the bouncing behavior on the UI.

<img width="309" alt="screen shot 2017-09-20 at 2 54 20 pm" src="https://user-images.githubusercontent.com/10248067/30669620-dec4bf84-9e13-11e7-8a01-eb6ba7bfab9e.png">

**Comment by rtibbles** [comment]
I think this is due to the fact that even importing a single channel is two different tasks - running `importchannel` and running `importcontent`.

**Comment by lyw07** [comment]
@jonboiser I realize that combining `importchannel` and `importcontent` progress bars from back-end is somewhat not really possible. To get the size of content to calculate the `total` for the overall progress bar, I need to query `LocalFile`, which are not accessible before running `importchannel`. @aronasorman suggested that I can get rid of `updateprogress` when calling `importchannel` so that users won't see the progress when doing `importchannel`.

**Comment by jonboiser** [comment]
@lyw07 That sounds like a reasonable idea. Does `importchannel` only import the Content metadata db? In that case, the file is generally very small and downloads almost instantly anyway.

**Comment by lyw07** [comment]
Yes, it's pretty small. More specifically, one call to `importchannel` only imports around 0.3% of all the data

**Comment by rtibbles** [comment]
The slight issue can be that it can take some time if there is a lot of metadata to import from the content db file.

**Comment by lyw07** [comment]
@rtibbles That's true.. But currently I'm not able to come up with an idea to get the content size before `importchannel` for the overall progress bar. Currently, my implementation is ignoring the progress of `importchannel` and creating a overall progress bar for content of multiple channels.

**Comment by jonboiser** [comment]
Can we have the Task have an indeterminate status like 'SCHEDULED' or 'QUEUED' (or maybe a brand new status) while reading the metadata? If a Task has such a status, the progress bar will still provide useful feedback to the user. Then the task can be in RUNNING status only once the content starts downloading.

**Comment by indirectlylit** [comment]
> indeterminate status

ref: [some thoughts](https://github.com/learningequality/iceqube/issues/20#issuecomment-326140507) on adding an "indeterminate progress" boolean flag as alternative to a percentage complete

**Comment by rtibbles** [comment]
I believe this will be resolved by the new import flow, as channel import and content import are now tracked separately.

**Comment by aronasorman** [comment]
Not a problem anymore, now that we have a separate flow for importing channels and importing content.
